### PR TITLE
Handle errors (more) correctly in bundle-cli

### PR DESCRIPTION
**Pull request by piotrswigon**
+ some tests

### CODE REVIEW

File: src/tools/bundle-cli.ts [hunk 1]
```diff
@@ -40,4 +40,7 @@ Examples
   process.exit();
 }
 
-void bundle(argv._, Array.isArray(argv.o) ? argv.o[0] : argv.o, argv.verbose);
+void bundle(argv._, Array.isArray(argv.o) ? argv.o[0] : argv.o, argv.verbose).catch(e => {
```
Comment (Cypher1) [inline src/tools/bundle-cli.ts]: Its not clear to me how this works. What is argv.o?

File: src/tools/tests/bundle-test.ts [hunk 2]
```diff
@@ -60,6 +60,21 @@ describe('Bundle Tool', () => {
...
+        .catch(e => done());
+  });
+  it('fails for a syntax error', done => {
+    bundle(['src/tools/tests/test-data/invalid/syntax-problem.manifest'], 'test-output/bundle/nope.zip', false)
+        .then(_ => assert.fail('should have failed'))
+        .catch(e => done());
+  });
+  it('fails for a non existent file reference', done => {
+    bundle(['src/tools/tests/test-data/invalid/reference-problem.manifest'], 'test-output/bundle/nope.zip', false)
+        .then(_ => assert.fail('should have failed'))
```
Comment (Cypher1) [inline src/tools/tests/bundle-test.ts]: Can we use `assert.throws` to check that the bundling is throwing a reasonable error?

File: src/tools/bundle-cli.ts (hunk 1 above)
Comment (piotrswigon) [reply src/tools/bundle-cli.ts]: I've changed it a bit, hopefully for better. `argv` holds arguments. argv.o === argv.output (-o vs --output) and specifies the path to the resulting bundle. PTAL.

File: src/tools/tests/bundle-test.ts (hunk 2 above)
Comment (piotrswigon) [reply src/tools/tests/bundle-test.ts]: assert.throws deals with synchronous code, while here I'm dealing with async one. I didn't find a way to utilize assert.throws here, let me know if you know how to do it.

I've added asserts to verify that the error thrown described the expected problem, thanks for the suggestion.

Comment (Cypher1) [reply src/tools/tests/bundle-test.ts]: Thanks :+1:

File: src/tools/bundle-cli.ts (hunk 1 above)
Comment (Cypher1) [reply src/tools/bundle-cli.ts]: I think we can put this down to my lack of experience with this api, I won't block on it anyway.

**Pull request merged**